### DISCUSSION TITLE
Check to see if I am running tar_make_clustermq or tar_make_future?

### DISCUSSION THREAD

**Discussion by PhDyellow** [ropensci/targets#605] [Help] [answered]
Thank you for the targets package, and drake before it! I'm using it in my PhD thesis to keep all the moving parts aligned, and give me the flexibility to move to the HPC once all the bugs are ironed out.

I am using a package in my pipeline that calls future.apply for internal parallelisation. The rest of the pipeline is fairly parallel, so I want to run the package with a sequential plan, and do all the parallelisation over target branches.

I understand that future plans can be nested. I noticed in discussion #359 that the recommended approach for using future within the pipeline under `tar_make_future` is to have a list of plans. `targets` will use the first plan, and any packages or targets within the pipeline will use the second plan. However, `tar_make_clustermq` has a completely different approach, where the future plan must be set inside a target. 

Does this mean that I have to set up my plan for either clustermq OR future? Is there a flag or test to check whether we were called with `tar_make`, `tar_make_clustermq` or `tar_make_future`?

**Comment by wlandau** [comment, marked as answer]
`targets` itself has no way of telling you it is running `tar_make_future()`, but there may be a hacky way of checking if you are running `future`. For me, `getOption("future.startup.script")` returns `FALSE` (maybe `TRUE` under other conditions) inside a future and `NULL` outside a future. Maybe ask at https://github.com/HenrikBengtsson/future/discussions to confirm if this is using `future` as it was intended to be used.

**Reply by PhDyellow** [reply]
Thanks for confirming that `targets` has no way of knowing which `tar_make_*()` was called. I will just build my pipeline around `tar_make_clustermq()`. I have my environment in a singularity container, so I can guarantee that `clustermq` is available, and also supporting `tar_make_future` sounds like too much of a hack.

I do like the idea of checking if we are in a future at the start of `_targets.R` as a rough test of whether the user has called `tar_make_future()` or not. I will keep that in mind.

Thanks for your help!

**Reply by wlandau** [reply]
I was thinking you would check it within a target. It you want to check something in _targets.R, maybe check that future::plan() does not return the sequential plan.
